src/System.Memory/src/System/Buffers/BuffersExtensions.cs

```diff
@@ -38,10 +58,33 @@ public static void CopyTo<T>(this ReadOnlySequence<T> sequence, Span<T> destinat
             if (sequence.Length > destination.Length)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.destination);
 
-            foreach (var segment in sequence)
+            if (sequence.IsSingleSegment)
+            {
+                sequence.First.Span.CopyTo(destination);
+            }
+            else
+            {
+                CopyToMultiSegement(sequence, destination);
+            }
+        }
+
+        private static void CopyToMultiSegement<T>(this ReadOnlySequence<T> sequence, Span<T> destination)
+        {
+            var enumerator = sequence.GetEnumerator();
```

Comment:
Is TryGet more efficient and the enumerator?

/cc @KrzysztofCwalina 😄

Reply:
Yea, as enumerator does TryGet plus filler